### DISCUSSION TITLE
S3 repository and Redis Global Cache and Local Storage Usage

### DISCUSSION THREAD

**Discussion by chezsmithy** [renovatebot/renovate#28453] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab

### Please tell us more about your question or problem

We are running into local storage limitations with our GitLab runners which are using Kubernetes. We are interested in implementing the s3 repository cache and Redis global cache to reduce the need for local storage on our Pods. I'm curious how the caching features actually work.

For the s3 repository cache, is it downloaded at the start of each run, and then uploaded at the end of each run; OR is the cache never written to local storage and instead directed completely to s3?

For the Redis global cache, is it downloaded at the start of each run, and then uploaded at the end of each run; OR is the cache never written to local storage and instead directed completely to Redis?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
For package cache using Redis, cache is not written locally. Instead Renovate reads and writes to Redis per cache item.

For s3 repository cache I'm not sure and not willing to commit to if there's no _temporary_ local storage during the run, but I don't think this matters. What matters is that such temporary storage is not persisted, so you don't end up with 100s or 1000s of repository caches locally if s3 cache is enabled. At most you'd have one repository's cache at a time and this is not particularly large - in the megabytes.
